File: backend/app/schemas/market_learning_progress.py
from datetime import datetime
from decimal import Decimal
from typing import Literal

from pydantic import BaseModel


class MarketLearningProgressRead(BaseModel):
    total_sessions: int
    completed_sessions: int
    unique_scenarios: int
    scenario_counts: dict[str, int]
    risk_tier_counts: dict[str, int]
    average_projected_return: Decimal
    latest_completed_at: datetime | None
    proficiency_level: Literal["not_started", "foundational", "developing", "proficient"]
    evidence_badge_eligible: bool
    next_learning_step: str
    disclaimer: str

File: frontend/lib/market-learning-progress.ts
export type MarketLearningProgress = {
  total_sessions: number;
  completed_sessions: number;
  unique_scenarios: number;
  scenario_counts: Record<string, number>;
  risk_tier_counts: Record<string, number>;
  average_projected_return: string;
  latest_completed_at: string | null;
  proficiency_level: "not_started" | "foundational" | "developing" | "proficient";
  evidence_badge_eligible: boolean;
  next_learning_step: string;
  disclaimer: string;
};
